faust/web/cache/cache.py

```diff
@@ -105,29 +117,26 @@ def _view_backend(self, view: View) -> CacheBackendT:
     def can_cache_request(self, request: Request) -> bool:
         return True
 
-    def can_cache_response(self,
-                           request: Request,
-                           response: Response) -> bool:
+    def can_cache_response(self, request: Request, response: Response) -> bool:
         return response.status == 200
 
     def key_for_request(self,
                         request: Request,
                         prefix: str = None,
-                        method: str = None) -> str:
+                        method: str = None,
+                        include_headers: bool = False) -> str:
         actual_method: str = request.method if method is None else method
+        headers = request.headers if include_headers else {}
         if prefix is None:
             prefix = self.key_prefix
-        return self.build_key(request, actual_method, prefix, [])
-
-    def build_key(self,
-                  request: Request,
-                  method: str,
-                  prefix: str,
-                  headers: List[str]) -> str:
-        context = hashlib.md5(
-            b''.join(v.encode() for v in headers if v is not None)).hexdigest()
-        url = hashlib.md5(
-            iri_to_uri(str(request.url)).encode('ascii')).hexdigest()
+        return self.build_key(request, actual_method, prefix, headers)
+
+    def build_key(self, request: Request, method: str, prefix: str,
+                  headers: Mapping[str, str]) -> str:
+        context = hashlib.md5(b''.join(
+            (k + v).encode() for k, v in headers.items())).hexdigest()
```

Comment:
changed this to ``want_bytes(k) + want_bytes(v) for k, v in headers.items())).hexdigest()``.
for the time something passes a `Dict[bytes, str]`